### PR TITLE
[16.0][MIG] shopinvader_image

### PR DESCRIPTION
**Pull request by lmignon**
* [x] depends on #1390
* [x] depends on https://github.com/OCA/search-engine/pull/158
* [x] depends on https://github.com/OCA/search-engine/pull/159

### CODE REVIEW

File: shopinvader_search_engine_image/tests/common.py [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+            }
+        )
+        cls.env["se.index"].create(
+            {
+                "name": "category",
+                "backend_id": cls.backend.id,
+                "model_id": cls.env.ref("product.model_product_category").id,
+                "serializer_type": "shopinvader_category_exports",
+            }
+        )
```
Comment (marielejeune) [inline shopinvader_search_engine_image/tests/common.py]:
```suggestion
```
Duplicate with lines 92-107

File: shopinvader_search_engine_image/schemas/shopinvader_category.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
+# Copyright 2023 ACSONE SA/NV
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo.addons.shopinvader_product.schemas.category import (
+    ShopinvaderCategory as BaseShopinvaderCategory,
```
Comment (marielejeune) [inline shopinvader_search_engine_image/schemas/shopinvader_category.py:5]:
```suggestion
    ShortShopinvaderCategory as BaseShopinvaderCategory,
```
?

File: shopinvader_search_engine_image/tests/__init__.py [hunk 3]
```diff
@@ -0,0 +1 @@
+from . import test_shopinvader_search_engine_image
```
Comment (marielejeune) [inline shopinvader_search_engine_image/tests/__init__.py:1]:
```suggestion
from . import test_shopinvader_search_engine_image
from . import test_images
```

File: shopinvader_search_engine_image/tests/common.py [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+
+        cls.loader.update_registry(
+            (
+                SeIndex,
+                SeBackend,
+            )
+        )
+        cls.backend = cls.env["se.backend"].create(
+            {"name": "Fake SE", "tech_name": "fake_se", "backend_type": "fake"}
+        )
```
Comment (marielejeune) [inline shopinvader_search_engine_image/tests/common.py:30]: I don't know when tests ran on GitHub, but they fail for me today, when creating the backend:

```
 File "/home/marie/.virtualenvs/odoo-shopinvader/lib/python3.10/site-packages/odoo/addons/shopinvader_search_engine_image/tests/common.py", line 28, in setUpClass
    cls.backend = cls.env["se.backend"].create(
  File "/home/marie/.virtualenvs/odoo-shopinvader/lib/python3.10/site-packages/decorator.py", line 232, in fun
    return caller(func, *(extras + args), **kw)
  File "/home/marie/.virtualenvs/odoo-shopinvader/lib/python3.10/site-packages/odoo/api.py", line 409, in _model_create_multi
    return create(self, [arg])
  File "/home/marie/.virtualenvs/odoo-shopinvader/lib/python3.10/site-packages/odoo/addons/server_environment/models/server_env_tech_name_mixin.py", line 64, in create
    return super().create(vals_list)
  File "/home/marie/.virtualenvs/odoo-shopinvader/lib/python3.10/site-packages/decorator.py", line 232, in fun
    return caller(func, *(extras + args), **kw)
  File "/home/marie/.virtualenvs/odoo-shopinvader/lib/python3.10/site-packages/odoo/api.py", line 410, in _model_create_multi
    return create(self, arg)
  File "/home/marie/.virtualenvs/odoo-shopinvader/lib/python3.10/site-packages/odoo/addons/base/models/ir_fields.py", line 665, in create
    recs = super().create(vals_list)
  File "/home/marie/.virtualenvs/odoo-shopinvader/lib/python3.10/site-packages/decorator.py", line 232, in fun
    return caller(func, *(extras + args), **kw)
  File "/home/marie/.virtualenvs/odoo-shopinvader/lib/python3.10/site-packages/odoo/api.py", line 410, in _model_create_multi
    return create(self, arg)
  File "/home/marie/.virtualenvs/odoo-shopinvader/lib/python3.10/site-packages/odoo/models.py", line 3967, in create
    next(iter(fields)).determine_inverse(batch_recs)
  File "/home/marie/.virtualenvs/odoo-shopinvader/lib/python3.10/site-packages/odoo/fields.py", line 1377, in determine_inverse
    determine(self.inverse, records)
  File "/home/marie/.virtualenvs/odoo-shopinvader/lib/python3.10/site-packages/odoo/fields.py", line 96, in determine
    if needle.__name__.find('__'):
AttributeError: 'functools.partial' object has no attribute '__name__'

```

File: shopinvader_search_engine_image/tests/__init__.py (hunk 3 above)
Comment (lmignon) [reply shopinvader_search_engine_image/tests/__init__.py:1]: I still need to cleanup the tests and `test_images` is the previous test file I need to rewrite.

File: shopinvader_search_engine_image/schemas/product_category.py [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+
+from odoo.addons.shopinvader_product.schemas import ProductCategory as BaseCategory
+
+from .images_mixin import ImageMixin
+
+
+class ProductCategory(BaseCategory, ImageMixin, extends=True):
+    @classmethod
+    def from_product_category(cls, odoo_rec):
+        obj = super().from_product_category(odoo_rec, with_parent=True, with_child=True)
```
Comment (qgroulard) [inline shopinvader_search_engine_image/schemas/product_category.py]: @lmignon Here is the bug on category bindings.

```suggestion
        obj = super().from_product_category(odoo_rec)
```

Comment (lmignon) [reply shopinvader_search_engine_image/schemas/product_category.py]: Thank you @qgroulard

**Pull request merged**